src/parse/asp/lexer.go

```diff
@@ -108,6 +99,34 @@ type lex struct {
 	lastEOL bool
 }
 
+// readNextByte updates the current and next byte fields by consuming from the underlying reader
+func (l *lex) readNextByte() {
+	var err error
+	l.currentRune = l.nextRune
+	l.nextRune, _, err = l.reader.ReadRune()
+
+	if err != nil {
+		if err != io.EOF {
+			fail(Position{Filename: l.filename, Offset: l.pos, Line: l.line, Column: l.col}, err.Error())
+		}
+
+		if l.currentRune != '\n' {
+			// If the file doesn't end in a newline, we will reject it with an "unexpected end of file"
+			// error. That's a bit crap so quietly fix it up here.
+			l.nextRune = '\n'
+		} else {
+			l.nextRune = 0 // This essentially null terminates the file which is useful later on
+		}
+	}
+}
+
+// advance reads moves to the next position in the reader, updating nextRune and currentRune
+func (l *lex) advance() {
+	l.col++
+	l.pos++
+	l.readNextByte()
```

Comment:
I think `readNextByte` should be named `readNextRune` since it operates on `ReadRune()` underneath.

Also, the size of a rune doesn't always equal 1 single byte. The `ReadRune()` function returns the number of bytes read which should be used when updating `pos`?

Reply:
Good catch. I refactored this lexer to be rune based without considering this.